docs/libraries/typespec-client-generator-core/rules/CONTRIBUTING.md

```diff
@@ -0,0 +1,36 @@
+# Contributing Linter Rules to the @azure-tools/typespec-client-generator-core Package
+
+For information about linter rules in typespec in general, view [here][generic-linter]
+
+In the `@azure-tools/typespec-client-generator-core` library, we have two main types of rules:
+
+1. Generic rule that applies to all emitted languages
+2. Specific rule that applies to a subset of all language: it can even apply to just one language
+
+The process for adding a rule starts off the same for both: for language-specific rules, there is an extra step
+
+1. Write the rule in `typespec-azure/packages/typespec-client-generator-core/src/rules/[rule-name].rule.ts
+2. Add reference to the rule in the `rules` array in [`typespec-azure/packages/typespec-client-generator-core/src/linter.ts`][tcgc-linter]
+   - This will automatically add it to a ruleset called `:all` for `@azure-tools/typespec-client-generator-core`
+3. Add the rule to the enable list for [`data-plane.ts`][data-plane-ruleset] and/or [`resource-manager.ts`][resource-manager-ruleset] in the [rulesets][rulesets] package. You can set `enable` to `false` here, if you want to delay enabling
+
+**If you are adding a language-specific rule**, you will also need this extra step 4. Add reference to the rule in the `[language]Rules` array in [`typespec-azure/packages/typespec-client-generator-core/src/linter.ts`][tcgc-linter]
+
+For Azure generations then, all rules, including all language-specific rules, will be run on the specs.
```

Comment:
I think we want to limit this to the Tier 1 languages not necessarily `all` languages?  Should we specify this?  I assume Go could add their own rules in here and those wouldn't apply unless generating Go even for azure?

Reply:
That's fair, let me update this to say that language-agnostic rules, and tier-1 languages will be run